pkg/resource/clickpipe.go

```diff
@@ -0,0 +1,1092 @@
+	if request.ProviderData == nil {
+		return
+	}
+
+	c.client = request.ProviderData.(api.Client)
+}
+
+func (c *ClickPipeResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = request.ProviderTypeName + "_clickpipe"
+}
+
+func (c *ClickPipeResource) Schema(_ context.Context, _ resource.SchemaRequest, response *resource.SchemaResponse) {
+	wrapStringsWithBackticksAndJoinCommaSeparated := func(s []string) string {
+		wrapped := make([]string, len(s))
+		for i, v := range s {
+			wrapped[i] = "`" + v + "`"
+		}
+		return strings.Join(wrapped, ", ")
+	}
+
+	response.Schema = schema.Schema{
+		MarkdownDescription: clickPipeResourceDescription,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the ClickPipe. Generated by the ClickHouse Cloud.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"service_id": schema.StringAttribute{
+				Description: "The ID of the service to which the ClickPipe belongs.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the ClickPipe.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"description": schema.StringAttribute{
+				Description: "The description of the ClickPipe.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"scaling": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"replicas": schema.Int64Attribute{
+						Description: "The number of desired replicas for the ClickPipe. Default is 1. The maximum value is 10.",
+						Optional:    true,
+						Computed:    true,
+						Default:     int64default.StaticInt64(1),
+						Validators: []validator.Int64{
+							int64validator.Between(1, 10),
+						},
+					},
+				},
+				Optional: true,
+			},
+			"state": schema.StringAttribute{
+				MarkdownDescription: "The state of the ClickPipe. (`Running`, `Stopped`). Default is `Running`. Whenever the pipe state changes, the Terraform provider will try to ensure the actual state matches the planned value. If pipe is `Failed` and plan is `Running`, the provider will try to resume the pipe. If plan is `Stopped`, the provider will try to stop the pipe. If the pipe is `InternalError`, no action will be taken.",
+				Optional:            true,
+				Default:             stringdefault.StaticString(api.ClickPipeRunningState),
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(api.ClickPipeRunningState, api.ClickPipeStoppedState),
```

Comment:
this confuses me a bit. you say valid values are "Running" and "Stopped" but then in the description you mention "Failed" and "InternalError" as well. Does it work at all?
What I mean, if the server pushes value "Failed" to the terraform client, will that be validated when storing the state or will the validation only apply to client-supplied  plan?

Reply:
Validator matches against user input. Yes, we expect the user to provide `Running` or `Failed` only.
However, ClickPipe can have various states, like `Failed`.  I've tested it with Terraform state, and it works fine.
Whenever API returns other status, Terraform consumes it and proposes update action accordingly. 

I will make the description a bit more descriptive.